packages/babel-generator/src/buffer.js

```diff
@@ -99,6 +98,19 @@ export default class Buffer {
   }
 
   endsWith(str: string): boolean {
+    // Fast path to avoid iterating over this._queue.
+    if (str.length === 1) {
+      let last;
+      if (this._queue.length > 0) {
+        const str = this._queue[0][0];
```

Comment:
Ah yeah that's confusing, I'll find a better name for it. It's not the same, just shadowing it on accident.